test cases/common/241 dependency native host == build/meson.build

```
@@ -0,0 +1,10 @@
+project('foo')
+
+dependency('zlib')
+dependency('zlib', native : true)
+dependency('zlib', native : false)
+
+# `native: true` should not make a difference when doing a native build.
+if not meson.is_cross_build()
+  dependency('expat', native: true)
```

oh missed that, why are you checking for expat?

```suggestion
  meson.override_dependency('something', declare_dependency())
  dependency('something', native: true)
```